python/xorbits/_mars/dataframe/base/pivot_table.py

```diff
@@ -0,0 +1,433 @@
+                    [proxy_chunk], shape=(np.nan, np.nan), index=out_idx
+                )
+            )
+
+        # generate output chunks
+        out_chunks = []
+        for chunk in reduce_chunks:
+            new_op = op.copy().reset_key()
+            new_shape = (np.nan, np.nan)
+            params = dict(shape=new_shape, index=chunk.index)
+            params.update(
+                dict(
+                    dtypes=in_df.dtypes,
+                    columns_value=in_df.columns_value,
+                    index_value=parse_index(None, chunk.key, proxy_chunk.key),
+                )
+            )
+            out_chunks.append(new_op.new_chunk([chunk], **params))
+        yield out_chunks
+
+        filtered_chunks = [chunk for chunk in out_chunks if chunk.shape[1] != 0]
+        for i, chunk in enumerate(filtered_chunks):
+            chunk._index = (i, 0)
+
+        all_dtype_index = set().union(
+            *[set(chunk.dtypes.index) for chunk in filtered_chunks]
+        )
+        dtypes_chunks = pd.concat([chunk.dtypes for chunk in filtered_chunks])
+        dtypes_chunks = dtypes_chunks.groupby(dtypes_chunks.index).first()
+
+        # generate combine chunks
+        combine_chunks = []
+        for chunk in filtered_chunks:
+            combine_op = op.copy().reset_key()
+            combine_op.stage = OperandStage.combine
+            combine_op.all_dtype_index = all_dtype_index
+            params = dict(
+                shape=(chunk.shape[0], len(all_dtype_index)), index=chunk.index
+            )
+            params.update(
+                dict(
+                    dtypes=dtypes_chunks,
+                    columns_value=in_df.columns_value,
+                    index_value=chunk.index_value,
+                )
+            )
+            combine_chunks.append(combine_op.new_chunk([chunk], **params))
+
+        new_op = op.copy()
+        kw = out_df.params.copy()
+        new_nsplits = (
+            tuple(chunk.shape[0] for chunk in combine_chunks),
+            (len(all_dtype_index),),
+        )
+        kw.update(dict(chunks=combine_chunks, nsplits=new_nsplits))
+
+        return new_op.new_tileables(op.inputs, **kw)
+
+    def __call__(self, df):
+        return self.new_dataframe([df])
+
+
+def df_pivot_table(
+    df: pd.DataFrame,
+    values: Any = None,
+    index: Any = None,
+    columns: Any = None,
+    aggfunc="mean",
+    fill_value=None,
+    margins: bool = False,
+    dropna: bool = True,
+    margins_name: str = "All",
+    observed: bool = False,
+    sort: bool = True,
+) -> pd.DataFrame:
+    """
+    Create a spreadsheet-style pivot table as a DataFrame.
+
+    The levels in the pivot table will be stored in MultiIndex objects
+    (hierarchical indexes) on the index and columns of the result DataFrame.
+
+    Parameters
+    ----------
+    values : list-like or scalar, optional
+        Column or columns to aggregate.
+    index : column, Grouper, array, or list of the previous
```

Comment:
Raise NotImplementedError on array and Grouper or list of them.

Reply:
done